backend/src/main/java/com/htecgroup/skynest/filter/CustomAuthorizationFilter.java

```diff
@@ -20,6 +22,12 @@
 @Log4j2
 public class CustomAuthorizationFilter extends OncePerRequestFilter {
 
+  // I'm using @Autowired because using constructor DI doesn't work.
+  // I think it has something to do with the way CustomAuthorizationFilter is defined as a bean in
+  // the application class, as it calls the NoArgsConstructor and the dependencies are not loaded.
+  // This way the service is autowired and it works.
+  @Autowired private InvalidJwtService invalidJwtService;
```

Comment:
if CustomAuthorizationFilter is defined as bean, you can inject InvalidJwtService in the class where CustomAuthorizationFilter is defined and pass that reference in the constructor. That way you can use the lombok @AllArgsConstructor and avoid Autowire annotation here.

Reply:
with other words:
 @Bean
  public CustomAuthorizationFilter customAuthorizationFilter(InvalidJwtService invalidJwtService) {
    return new CustomAuthorizationFilter(invalidJwtService);
  }
and use standard constructor injection here with Lombok.

On the other hand you can create  new configuration class (annotated with @Configuration) and move there all beans defined in SkyNestApplication.class. That way you can separate the bean creation logic from application start

Reply:
Yes, it was already done in another PR, so I'am just gonna use it now.